plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/StdLibConverter.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.formver.conversion
+
+import org.jetbrains.kotlin.fir.symbols.impl.FirFunctionSymbol
+import org.jetbrains.kotlin.formver.embeddings.callables.FunctionSignature
+import org.jetbrains.kotlin.formver.isCollection
+import org.jetbrains.kotlin.formver.viper.ast.Exp
+import org.jetbrains.kotlin.formver.viper.ast.Exp.*
+
+fun LocalVar.sameSize(): Exp = EqCmp(fieldAccess(SpecialFields.ListSizeField), Old(fieldAccess(SpecialFields.ListSizeField)))
+fun LocalVar.increasedSize(amount: Int): Exp =
+    EqCmp(fieldAccess(SpecialFields.ListSizeField), Add(Old(fieldAccess(SpecialFields.ListSizeField)), IntLit(amount)))
+
+fun FirFunctionSymbol<*>.stdLibPreConditions(signature: FunctionSignature): List<Exp> =
```

Comment:
Would it be possible to handle this using `SpecialFunctions` instead of this approach?  I'm not necessarily opposed to doing it with this for now, but why don't those work?

Reply:
using `SpecialFunctions` was my initial approach, but there were some issues regarding polymorphism that are solved for free with this approach. Switching to `SpecialFunctions` will require more work, I think that if there are not any clear benefit in doing it it, this is not necessary